arch/platform/cooja/sys/cooja-log.c

```diff
@@ -124,41 +121,26 @@ putchar(int c)
       simlog_char(SLIP_END);
       debug_frame = 0;
     }
-
-    return c;
   } else {
     simlog_char(c);
-    return c;
   }
+  return c;
 }
 /*-----------------------------------------------------------------------------------*/
-int
-puts(const char* s)
-{
-  simlog(s);
-  simlog_char('\n');
-  return 0;
-}
-/*-----------------------------------------------------------------------------------*/
-int
-printf(const char *fmt, ...)
+unsigned int
+dbg_send_bytes(const unsigned char *s, unsigned int len)
 {
-  int res;
-  static char buf[MAX_LOG_LENGTH];
-  va_list ap;
-  int i;
+  unsigned int i = 0;
 
-  va_start(ap, fmt);
-  res = vsnprintf(buf, MAX_LOG_LENGTH, fmt, ap);
-  va_end(ap);
-
-  //    simlog(buf);
-  for(i = 0; i < res; i++) {
-    putchar(buf[i]);
+  while(s && *s != 0) {
+    if(i >= len) {
```

Comment:
Can this if-statement be merged into the while() condition on the previous line?

Reply:
Good point. I have simplified the loop.